### ISSUE TITLE
[FEATURE] check Headers on read first row of file CSV

### ISSUE THREAD

**Issue by lytaitruong** [C2FO/fast-csv#321]
- [ ] Formatting
 - [x] Parsing 

Describe event:
- I have the file stream CSV from S3 and I have to read It by your API. 
The file Csv will have:
1. First row: name & type each column
2. Second row: Data on each column 

The problem is:
- Instead using `event.on('data')` to check format the all database, it make i considers when the large row. I want just only using like `event.on('headers')` to return array first row to handle error before read all database.

**Describe the solution you'd like**
I see the library like 'csvtojson' or 'csv-parser' they have the `event.on('header')` to take the array of first row. If have it i can check and response for user the error.csv immediately to they fix
Or can option headers will matching the rule like custom Headers i see
`csv.parse({ headers: (header, index) => if(header or index is invalid) => handle this row`

Thank for your service

**Comment by doug-martin** [comment]
Thanks for the feature request! This is published under `v4.1.0`

**Comment by kaustavbhattacharjee** [comment]
Can I use the event.on('headers') in the fromPath function?
Example:

```
.on('headers',function(d){
        console.log(d);
      })
```
